test/framework/src/main/java/org/elasticsearch/index/shard/RestoreOnlyRepository.java

```diff
@@ -63,8 +64,7 @@ public RepositoryMetadata getMetadata() {
     }
 
     @Override
-    public SnapshotInfo getSnapshotInfo(SnapshotId snapshotId) {
-        return null;
+    public void getSnapshotInfo(GetSnapshotInfoContext context) {
```

Comment:
Doing nothing here leaks the context. Should we complete it (maybe with a UOE) or assert that this method isn't called?

Reply:
++ made it throw